### PR TITLE
Draft of #54: Configurable extensions

### PR DESCRIPTION
**Pull request by RhysU**
Please review/test carefully-- I'm new to parinfer, lisp-like languages, and VimScript.

### CODE REVIEW

File: plugin/parinfer.vim [hunk 1]
```diff
@@ -3,9 +3,19 @@
...
+    \ 'mode':       "indent",
+    \ 'script_dir': resolve(expand("<sfile>:p:h:h"))
+    \ }
+lockvar! g:_VIM_PARINFER_DEFAULTS
+
+for s:key in keys(g:_VIM_PARINFER_DEFAULTS)
+    if !exists('g:vim_parinfer_' . s:key)
+        let g:vim_parinfer_{s:key} = copy(g:_VIM_PARINFER_DEFAULTS[s:key])
+    endif
+endfor
```
Comment (RhysU) [inline plugin/parinfer.vim]: Pattern lifted from syntastic.

File: plugin/parinfer.vim [hunk 2]
```diff
@@ -151,23 +161,21 @@ com! -bar ToggleParinferMode cal parinfer#ToggleParinferMode()
...
-  autocmd FileType clojure,racket,lisp,scheme nnoremap <buffer> <Tab> :call parinfer#do_indent()<cr>
-  autocmd FileType clojure,racket,lisp,scheme nnoremap <buffer> <Tab> :call parinfer#do_indent()<cr>
-  autocmd FileType clojure,racket,lisp,scheme nnoremap <buffer> <S-Tab> :call parinfer#do_undent()<cr>
-  autocmd FileType clojure,racket,lisp,scheme vnoremap <buffer> <Tab> :call parinfer#do_indent()<cr>
-  autocmd FileType clojure,racket,lisp,scheme vnoremap <buffer> <S-Tab> :call parinfer#do_undent()<cr>
+  execute "autocmd InsertLeave " . join(g:vim_parinfer_globs, ",") . " call parinfer#process_form()"
+  execute "autocmd FileType " . join(g:vim_parinfer_filetypes, ",") . " nnoremap <buffer> <Tab> :call parinfer#do_indent()<cr>"
+  execute "autocmd FileType " . join(g:vim_parinfer_filetypes, ",") . " nnoremap <buffer> <S-Tab> :call parinfer#do_undent()<cr>"
+  execute "autocmd FileType " . join(g:vim_parinfer_filetypes, ",") . " vnoremap <buffer> <Tab> :call parinfer#do_indent()<cr>"
+  execute "autocmd FileType " . join(g:vim_parinfer_filetypes, ",") . " vnoremap <buffer> <S-Tab> :call parinfer#do_undent()<cr>"
```
Comment (RhysU) [inline plugin/parinfer.vim]: There may well be better/cleaner ways to avoid the repeated joins.

**Comment by bhurlow** [comment]
@RhysU yea this looks 💯, I'm fine with some repetitions in the joins there since it's very concrete. LMK if this is ready to merge and I will do so

**Comment by RhysU** [comment]
@bhurlow, I fixed a couple of typos in the documentation-- please merge away whenever you're hapy.

File: plugin/parinfer.vim (hunk 2 above)
Comment (RhysU) [reply plugin/parinfer.vim]: Turns out there is a good pattern, just a `let s:foo = expr` statement.  Still, leaving as-is for now.

**Pull request merged**